### ISSUE TITLE
[ISSUE] It Keeps evolving instead of keep to use luck egg.

### ISSUE THREAD

**Issue by vcafezinho** [NECROBOTIO/NecroBot#3811]
No matter how I configure, it keeps evolving instead wait to use luck eggs.

If I switch
  "EvolveAllPokemonWithEnoughCandy": false,
  "EvolveKeptPokemonsAtStorageUsagePercentage": 0.0,
It will evolves anyway. Without waiting for the number that I set   "UseLuckyEggsMinPokemonAmount": 30,

If:
  "EvolveAllPokemonWithEnoughCandy": true,
  "EvolveKeptPokemonsAtStorageUsagePercentage": 0.0,
It will evolves anyway. Without waiting for the number that I set   "UseLuckyEggsMinPokemonAmount": 30,

If:
  "EvolveAllPokemonWithEnoughCandy": true,
  "EvolveKeptPokemonsAtStorageUsagePercentage": 9.0,
It will wait until percentage. However without take in consideration UseLuckyEggsMinPokemonAmount": 30, If you have less than 30 pokes to evolve it will not use the eggs and evolve all possible.

Maybe what I am looking is not implemented. What I want is set up a number of pokemons to be keeped and use luck egg and evolve them all. Not taking in consideration those two options:
  "EvolveAllPokemonWithEnoughCandy": true,
  "EvolveKeptPokemonsAtStorageUsagePercentage": 0.0,

Example:
  "EvolveAllPokemonWithEnoughCandy": false,
  "EvolveKeptPokemonsAtStorageUsagePercentage": 0.0,
  "KeepPokemonsThatCanEvolve": true,
  "UseLuckyEggsMinPokemonAmount": 30,
  "UseLuckyEggsWhileEvolving": true,
Does not matter how many pokes you have in your bag, it will wait until you have 30 pokes with their respective candies to evolve all together after use luck egg. Pokemons to evolve and to keep to evolve based on "PokemonsToEvolve" list.

If there is a way or I am doing something wrong, please let me know.
Sorry my bad English.

Thanks

**Comment by PokeBotUK** [comment]
I think what he means is this is happening:

![image](https://cloud.githubusercontent.com/assets/20753079/17518161/99e0b834-5e3e-11e6-9011-a0111e435f99.png)


Even though its supposed to wait for 4 to evolve and then use lucky egg - it just eveolves them and ignores the settings.

**Comment by FranManre** [comment]
maybe bot is failing there, take a look on this, maybe it helps: https://github.com/NECROBOTIO/NecroBot/wiki/Config#evolution

**Comment by ChemiHatake** [comment]
Oh wait, I've fixed it but someone changed it back because he didn't like my logic.
great

**Comment by Botter22** [comment]
Could there be function to use lucky egg when inventory is full to start evolving?

**Comment by weedorbeat** [comment]
@ChemiHatake you will revert it?

**Comment by ChemiHatake** [comment]
@weedorbeat no, I'm totally done with this repo lol.

**Comment by kauegimenes** [comment]
This was fixed?

**Comment by weedorbeat** [comment]
@ChemiHatake It is a pity to hear that :(
@kauegimenes I think so... #3388 and #3296, the changes are not to be found in the current code. Maybe @ChemiHatake can correct me if I'm wrong

**Comment by actislav** [comment]
"EvolveAllPokemonWithEnoughCandy": true,
"EvolveKeptPokemonsAtStorageUsagePercentage": 95.0,
"KeepPokemonsThatCanEvolve": true,
"UseLuckyEggsMinPokemonAmount": 30,
"UseLuckyEggsWhileEvolving": true,

use this, if have enough room to have 30 pokemon to evolve at once it will work as u need, but if not it means it impossible with this settings and your storage

**Comment by ChemiHatake** [comment]
@weedorbeat Yes, it is correct.
my code (that actually worked) was reverted

**Comment by vcafezinho** [comment]
@ChemiHatake You are completely right, it was working before. So as I see nobody besides you wanted to fix that, when you got it fixed another person changed again your code that was working. Now we have an option that actually is a mess and do not work as it should be =/

**Comment by ChemiHatake** [comment]
@vcafezinho Its just he didn't like my logic, (my logic was correct according to the wiki) so he changed it to match his logic.

**Comment by vcafezinho** [comment]
@ChemiHatake And who did this is wrong. Change something that was working to a fully bullshit???

**Comment by ChemiHatake** [comment]
@vcafezinho not quite sure,
but that's one of the reasons I stopped contributing,
the number of devs who actually read the wiki is very close to zero,
and what that creates, is a big amount of devs trying to make the code match their logic instead of matching the code to the intended logic. (and the intended logic **is in the wiki**)

they don't actually check things deeply or think again before they close a pull request someone was working hard on. 
this is just my experience here so far.

**Comment by 20giis** [comment]
YES MAN IT ACTUALLY WORKED BEFORE, LIKE PERFECTLY, I'VE BEEN TRYING TO FIGURE OUT THIS BULLSHIT

**Comment by 20giis** [comment]
Oh and thanks Chemi, it was amazing while it lasted! <3

**Comment by bduff9** [comment]
I see the same issue.  Also, seems like which pokeball gets used is not totally working (i.e. it uses a great ball for a basic pidgey that didn't meet the criteria for using a great ball), but I need to test more before I open a new issue
